src/txmempool.h

```diff
@@ -67,19 +83,26 @@ class CTxMemPoolEntry
     int64_t feeDelta;          //!< Used for determining the priority of the transaction for mining in a block
     uint32_t nBranchId;        //!< Branch ID this transaction is known to commit to, cached for efficiency
 
+    // Information about descendants of this transaction that are in the
+    // mempool; if we remove this transaction we must remove all of these
+    // descendants as well.  if nCountWithDescendants is 0, treat this entry as
+    // dirty, and nSizeWithDescendants and nFeesWithDescendants will not be
+    // correct.
+    uint64_t nCountWithDescendants;  //! number of descendant transactions
```

Comment:
nb, but maybe for a followup PR: The treatment of 0 here suggests that these three fields should all be part of a std::optional<SomeStruct> that contains these field values.